fuel/datasets/hdf5.py

```diff
@@ -411,21 +616,27 @@ def _in_memory_get_data(self, state=None, request=None):
         return data, shapes
 
     def _out_of_memory_get_data(self, state=None, request=None):
+        if not isinstance(request, (slice, list)):
```

Comment:
The problem is, where do we do that? Not all datasets will support slices and/or indices, and there's no single base class for those that do I think, but we don't want to re-implement that check for every single dataset.